### PR TITLE
[New model] Add PETR

### PR DESCRIPTION
**Pull request by LielinJiang**

### CODE REVIEW

File: docs/models/petr/README.md [hunk 1]
```diff
@@ -0,0 +1,248 @@
...
+PETR...
+
+## <h2 id="3">训练配置</h2>
+
+我们提供了在开源数据集上的训练配置与结果，详见[PETR训练配置](../../../configs/petr)
+
+
+## <h2 id="4">模型库</h2>
+| 模型 |  骨干网络  | 3DmAP | NDS |  模型下载 | 配置文件 |  日志 |
+| :--: | :-------: | :--------: | :-------------------: | :--------------------------: | :-----------------------: | :------: | :-----: | :--: |
```
Comment (nepeplwu) [inline docs/models/petr/README.md]: 缺少vdl service的文件

File: paddle3d/datasets/nuscenes/nuscenes_multi_view_det.py [hunk 2]
```diff
@@ -0,0 +1,687 @@
...
+    """
+    DATASET_NAME = "Nuscenes"
+
+    def __init__(self,
+                 dataset_root: str,
+                 ann_file: str = None,
+                 mode: str = "train",
+                 transforms: Union[TransformABC, List[TransformABC]] = None,
+                 max_sweeps: int = 10,
+                 class_balanced_sampling: bool = False,
```
Comment (nepeplwu) [inline paddle3d/datasets/nuscenes/nuscenes_multi_view_det.py]: 似乎未用到

File: paddle3d/transforms/base.py [hunk 3]
```diff
@@ -45,5 +47,8 @@ def __call__(self, sample: Sample):
         if sample.modality == 'image' and sample.meta.channel_order == 'hwc':
             sample.data = sample.data.transpose((2, 0, 1))
             sample.meta.channel_order = "chw"
-
+        elif sample.modality == 'multiview':
+            sample.img = np.stack(
+                [img.transpose(2, 0, 1) for img in sample.img], axis=0)
+            # print('sample.img', sample.img.shape)
```
Comment (nepeplwu) [inline paddle3d/transforms/base.py]: 冗余的注释，建议去掉

File: paddle3d/transforms/transform.py [hunk 4]
```diff
@@ -312,3 +320,520 @@ def __call__(self, sample: Sample):
...
+                results["lidar2img"][view]).astype('float32') @ rot_mat_inv
+            results["extrinsics"][view] = np.array(
+                rot_mat_inv.T @ results["extrinsics"][view]).astype('float32')
+
+        results.gt_bboxes_3d[:, :6] *= scale_ratio
+        results.gt_bboxes_3d[:, 7:] *= scale_ratio
+        return
+
+
+def imnormalize(img, mean, std, to_rgb=True):
```
Comment (nepeplwu) [inline paddle3d/transforms/transform.py]: 类似代码是否应该放到functional中

File: paddle3d/datasets/nuscenes/nuscenes_multi_view_det.py [hunk 5]
```diff
@@ -0,0 +1,687 @@
...
+        max_sweeps (int, optional): Max number of sweeps. Default: 10.
+
+    Returns:
+        tuple[list[dict]]: Information of training set and validation set
+            that will be saved to the info file.
+    """
+    train_nusc_infos = []
+    val_nusc_infos = []
+
+    for sample in tqdm(nusc.sample):
```
Comment (nepeplwu) [inline paddle3d/datasets/nuscenes/nuscenes_multi_view_det.py]: 建议使用 `logger.range`
https://github.com/PaddlePaddle/Paddle3D/blob/develop/paddle3d/utils/logger.py#L154
https://github.com/PaddlePaddle/Paddle3D/blob/develop/paddle3d/datasets/generate_gt_database.py#L51

File: deploy/petr/python/infer.py [hunk 6]
```diff
@@ -0,0 +1,235 @@
...
+    Returns:
+        image [np.ndarray(H, W, 3)]: RGB Image
+    """
+    img = np.stack([cv2.imread(name) for name in filenames], axis=-1)
+    imgs = [img[..., i] for i in range(img.shape[-1])]
+
+    new_imgs = []
+    for i in range(len(imgs)):
+        img = cv2.resize(imgs[i], (800, 450), cv2.INTER_LINEAR)
+        img = img[130:, :, :]
```
Comment (nepeplwu) [inline deploy/petr/python/infer.py]: 为什么需要这么写？

Comment (LielinJiang) [reply deploy/petr/python/infer.py]: optimized. remove hard-code here.

File: paddle3d/datasets/nuscenes/nuscenes_multi_view_det.py (hunk 5 above)
Comment (LielinJiang) [reply paddle3d/datasets/nuscenes/nuscenes_multi_view_det.py]: done

File: paddle3d/transforms/transform.py (hunk 4 above)
Comment (LielinJiang) [reply paddle3d/transforms/transform.py]: done

File: paddle3d/transforms/base.py (hunk 3 above)
Comment (LielinJiang) [reply paddle3d/transforms/base.py]: removed

File: paddle3d/datasets/nuscenes/nuscenes_multi_view_det.py (hunk 2 above)
Comment (LielinJiang) [reply paddle3d/datasets/nuscenes/nuscenes_multi_view_det.py]: removed

File: docs/models/petr/README.md (hunk 1 above)
Comment (LielinJiang) [reply docs/models/petr/README.md]: added

**Pull request merged**